### PR TITLE
add-eks

### PR DESCRIPTION
**Pull request by calvix**
towards https://github.com/giantswarm/roadmap/issues/2820

pretty much copy/past from capa provider and some renaming
only change was to check worker nodes instead of control plane nodes as EKS ahs no control plane nodes
```
wait.AreNumNodesReady(state.GetContext(), client, 2, &cr.MatchingLabels{"node-role.kubernetes.io/worker": ""}),
```

### Checklist

- [ ] Update changelog in CHANGELOG.md.

### Trigger e2e tests

/run cluster-test-suites

### CODE REVIEW

**Comment by AverageMarcus** [comment]
@calvix I've updated this branch with a few small changes and bumped the clustertest version with the changes I've made to support EKS. Triggering tests to see what's left to tackle...

/run cluster-test-suites

**Comment by calvix** [comment]
nice

**Comment by AverageMarcus** [comment]
cert-manager seems to be failing to install correctly. 😞

**Comment by calvix** [comment]
When I create EKS cluster manually all apps were successfully deployed :disappointed: 

I still need to wait until this PR is merged to have working PVC tests https://github.com/giantswarm/cluster-eks/pull/33

**Comment by calvix** [comment]
:crossed_fingers: 
/run cluster-test-suites

**Comment by calvix** [comment]
the only failed tests are for CAPV and CAPZ . EKS works

**Comment by calvix** [comment]
the only failed tests are for CAPV and CAPZ . EKS works

**Comment by calvix** [comment]
@AverageMarcus @giantswarm/team-phoenix please take a look and approve if it works for you

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,19 +9,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Added
 
-- Add test for autoscaling MachinePool nodes.
```
Comment (AverageMarcus) [inline CHANGELOG.md:12]: I don't think these lines should have been removed, right?

File: internal/common/basic.go [hunk 2]
```diff
@@ -42,6 +42,11 @@ func runBasic() {
 			replicas, err := state.GetFramework().GetExpectedControlPlaneReplicas(state.GetContext(), state.GetCluster().Name, state.GetCluster().GetNamespace())
 			Expect(err).NotTo(HaveOccurred())
 
+			// Skip this test is the cluster is a managed cluster (e.g. EKS)
+			if replicas == 0 {
+				Skip("Bastion is not supported.")
```
Comment (AverageMarcus) [inline internal/common/basic.go:47]:
```suggestion
				Skip("ControlPlane is not supported.")
```

**Review by AverageMarcus — APPROVED** [review]
Just a couple small points but looks good other than that

File: CHANGELOG.md (hunk 1 above)
Comment (calvix) [reply CHANGELOG.md:12]: yeah

**Review by AverageMarcus — APPROVED** [review]

**Pull request merged**